pact/verifier.py

```diff
@@ -28,68 +28,160 @@ def __str__(self):
+            **kwargs
+    ):
         """Use Broker to verify.
 
-        Args:
-            broker_username ([String]): broker username
-            broker_password ([String]): broker password
-            broker_url ([String]): url of broker
-            enable_pending ([Boolean])
-            include_wip_pacts_since ([String])
-            publish_version ([String])
-
+        Usage:
+
+        >>> from pact import Verifier
+        >>>
+        >>> verifier = Verifier(
+        ...   provider='UserService',
+        ...   provider_base_url='http://localhost:5001'
+        ... )
+        >>>
+        >>> result, _ = verifier.verify_with_broker(
+        ...   'https://broker/consumer-provider.json',  # Optional
+        ...   enable_pending=True,
+        ...   broker_username='pactbroker',
+        ...   broker_password='pactbroker',
+        ...   broker_url='http://localhost',
+        ... )
+        >>>
+        >>> assert result == 0
+
+        :param pacts: Optional list of pacts to verify. Every pact in that list
+            can be an HTTP URI or a local file.
+        :type pacts: str
+        :param enable_pending: Allow pacts which are in pending state to be
+            verified without causing the overall task to fail.
+        :type enable_pending: bool
+        :param include_wip_pacts_since: Allow pacts that don't match given
+            consumer selectors (or tags) to  be verified, without causing the
+            overall task to fail
+        :type include_wip_pacts_since: str or None
+        :keyword str broker_username: Pact Broker basic auth username.
+        :keyword str broker_password: Pact Broker basic auth password.
+        :keyword str broker_token: Pact Broker bearer token.
+        :keyword str broker_url: Base URL of the Pact Broker from which to
+            retrieve the pacts.
+        :raises Exception: When `publish_verification_results` is set,
+            but `publish_version` is not provided.
+        :return: Returns a tuple of two elements. The first indicates the
+            status of the operation, so that 0 means success, and the second
+            contains the logs of the operation.
+        :rtype: tuple
         """
-        broker_username = kwargs.get('broker_username', None)
-        broker_password = kwargs.get('broker_password', None)
-        broker_url = kwargs.get('broker_url', None)
-        broker_token = kwargs.get('broker_token', None)
+        self.validate_publish(**kwargs)
 
         options = {
-            'broker_password': broker_password,
-            'broker_username': broker_username,
-            'broker_token': broker_token,
-            'broker_url': broker_url
+            'broker_password': kwargs.get('broker_password', None),
+            'broker_username': kwargs.get('broker_username', None),
+            'broker_token': kwargs.get('broker_token', None),
+            'broker_url': kwargs.get('broker_url', None),
         }
```

Comment:
I think all these args could go into `extract_params` along with the others and avoid needing the two steps for options/update options, both `verify` and `verify_with_broker` then go onto do the same approach to pull out the options and use if they are provided or not

Reply:
I don't want to change this API somehow in the current Pull Request. I've put a bit of cleanup here, it's a code cleanup and nothing more. Please, let's do such things in a separate Pull Request,  and here I don't even want to discuss this issue, because it's offtopic